src/test/java/ui/tests/AuthTest.java

```diff
@@ -0,0 +1,193 @@
+    Map<String, UserAccount> users = new HashMap<>();
+    UserAccountsPool userAccountsPool = new UserAccountsPool(users);
+    Map<String, UserAccountResult> usersUpdated = new HashMap<>();
+
+    @BeforeClass
+    public static void setup() {
+        logger.info("Setup process started");
+        System.setProperty("webdriver.chrome.driver", ConfigProperties.getProperty("chromedriver"));
+        driver = new ChromeDriver();
+        logger.info("Read user from Json");
+    }
+
+    @Before
+    public void init() {
+        logger.info("MainPage initiated");
+        mainPage = new MainPage(driver);
+        logger.info("AuthenticationPage initiated");
+        authenticationPage = new AuthenticationPage(driver);
+        logger.info("Account initiated");
+        accountPage = new AccountPage(driver);
+        logger.info("AccountCreationPage initiated");
+        accountCreationPage = new AccountCreationPage(driver);
+        logger.info("PersonalInformationPage initiated");
+        personalInformationPage = new PersonalInformationPage(driver);
+        logger.info("Setup implicitly wait for each method");
+        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+        driver.get(ConfigProperties.getProperty("mainpage"));
+    }
+
+    @Test
+    public void testSuccessfulRegistration() {
+        logger.info("Email generation");
+        String email = "flynn+" + RandomStringUtils.randomAlphanumeric(10) + "@gmail.com";
+        logger.info("Password generation");
+        String password = RandomStringUtils.randomAlphanumeric(10);
+        logger.info("Postal code generation");
+        String postCode = RandomStringUtils.randomNumeric(5);
+        logger.info("Mobile phone generation");
+        String mobilePhone = "+" + RandomStringUtils.randomNumeric(10);
+        logger.info("Opening main page and clicking Sign in button");
+
+        mainPage.clickSignInButton();
+        logger.info("Starting account registration with generated email");
+
+        authenticationPage.startAccountAuth(email);
+
+        logger.info("Read User Object to Json");
+        UserAccount user = new UserAccount("Johnny", "Flynn", password, email, "14", "3", "1983",
+                "South Africa", "Johannesburg",
+                "10", postCode, mobilePhone, "Home");
+        userAccountsPool.serialize("src/test/resources/accountData.json", user);
+
+        logger.info("Create expected account");
+        UserAccountResult expectedAccount =
+                new UserAccountResult("Johnny", "Flynn", email, "14", "3", "1983");
+
+        logger.info("Read Json to User Object");
+        userAccountsPool.deserialize("src/test/resources/accountData.json");
+
+        logger.info("Get User from User Account Pool");
+        UserAccount userPool = userAccountsPool
+                .getUser("src/test/resources/accountData.json");
+
+        logger.info("Account registration");
+        accountCreationPage.createAccount(userAccountsPool
```

Comment:
There's a variable `userPool` (which should be `userData` or `userAccount`) where the content of specified file is already stored